### DISCUSSION TITLE
Feature request : settings for action before projection and after removing projection

### DISCUSSION THREAD

**Discussion by coucout** [AntonyCorbett/OnlyM#464] [Ideas]
Hello,

I've made a similar software (not published for the moment) named "Projecteur". I prefer yours 😉 but it lacks of a function that I've programmed to automatize zoom share : a "configurable action" launched after the projection and a "configurable action" launched after the picture is removed from the projection.

I explain. I've configured zoom to begin share with Ctrl+F11. So I've made an autohotkey that fire these things:

- Send Ctrl+F11
- wait for 0.5 seconds (the time the share windows of zoom appear on the screen)
- find the picture on the screen (with ImageSearch()) (see screnshot above)
- double click to start the share

![image](https://user-images.githubusercontent.com/3479885/194703769-5b9b1342-6a95-4b46-9525-ad98680a94e5.png)

In my software, I've putted a setting to start specific action after the slide is displayed on the second screen.

If configured, i remove topmost attribute of "Projecteur", display the slide on the second screen and launch specific action.

After the slide is removed of the second screen, I launch specific action ( simply {^F11} to end zoom sharing).

Above, a screenshot of an extract of the settings. In french sorry ;-)
![image](https://user-images.githubusercontent.com/3479885/194704106-964340a7-29fe-4973-b355-346c7b0ed409.png)

**Comment by AntonyCorbett** [comment]
@coucout Thank you - that sounds like an interesting idea. Be sure to let us all know when you make your application available!

**Reply by coucout** [reply]
Perhaps I can try myself to implement this in onlyM. I have some experience with C#... Wich version have I to use ?

**Comment by AntonyCorbett** [comment]
@coucout The current solution uses .NET 5 (C# 9.0)
